src/Compilers/CSharp/Portable/Symbols/Source/SourceModuleSymbol.cs

```diff
@@ -322,6 +324,80 @@ private void ValidateLinkedAssemblies(BindingDiagnosticBag diagnostics, Cancella
             }
         }
 
+        internal void DiscoverInterceptorsIfNeeded()
+        {
+            if (!Volatile.Read(ref DeclaringCompilation.InterceptorsDiscoveryComplete))
+            {
+                discoverInterceptors();
+                Volatile.Write(ref DeclaringCompilation.InterceptorsDiscoveryComplete, true);
+            }
+
+            void discoverInterceptors()
+            {
+                var toVisit = ArrayBuilder<NamespaceOrTypeSymbol>.GetInstance();
+
+                var location = this.GlobalNamespace.GetFirstLocationOrNone();
+                if (!location.IsInSource)
+                {
+                    return;
```

Comment:
How do we get there? Is this for a compilation without any source?

Reply:
Perhaps, in this case maybe the global namespace is created but it doesn't have any locations. I'll add a test.